src/domain/task-system/domain/src/entities/task_status.rs

```diff
@@ -57,9 +57,23 @@ pub struct TaskUpdateDatasetResult {
     pub pull_result: PullResult,
 }
 
-impl From<PullResult> for TaskUpdateDatasetResult {
-    fn from(value: PullResult) -> Self {
-        Self { pull_result: value }
+impl TaskUpdateDatasetResult {
+    pub fn from_pull_result(pull_result: &PullResult, fetch_uncacheable: bool) -> Self {
+        match pull_result {
+            PullResult::Updated { .. } => TaskUpdateDatasetResult {
+                pull_result: pull_result.clone(),
+            },
+            PullResult::UpToDate(up_to_date_result) => match up_to_date_result {
+                UpToDateResult::UpToDate => TaskUpdateDatasetResult {
+                    pull_result: pull_result.clone(),
+                },
+                UpToDateResult::IngestUpToDate { uncacheable } => TaskUpdateDatasetResult {
+                    pull_result: PullResult::UpToDate(UpToDateResult::IngestUpToDate {
+                        uncacheable: *uncacheable && fetch_uncacheable,
```

Comment:
I don't understand why we need this logic. Task should wrap result from the lower level service, but here we are changing the meaning of `uncacheable` flag based on whether `fetch_uncacheable` parameter was specified.

Reply:
We have the same logic in the pull command. It looks reasonable to indicate it was flag provided or not to return have we tried to fetch_uncacheable flag or not.

Reply:
In the `PullCommand` this logic is present to not complain/warn about uncacheable sources if `--fetch-uncacheable` was already specified - no point to hint user about this flag if they are already using it.

But take into account that `PullCommand` is a UI/presentation layer - its logic is about how to present things that happen in the core most informatively to the user.

In your case we are in the core domain layer. Its job is not to decide what to show to the user but to provide detailed information to the upper layers about what actually happened under the hood.

When you apply this logic - this looks to me as changing the meaning of what `uncacheable` flag means.

I think the best solution is leave this filtering to UI assuming UI will have access to both:
- the task/flow result - that tells whether the source is uncacheable or not
- and the input parameter - whether `fetch-uncachable` was specified